menas/ui/service/RunService.js

```diff
@@ -0,0 +1,226 @@
+    Functions.ajax("api/runs", "GET", {},
+      oData => {
+        this._bindRunSummaries(oData, oMasterPage);
+        if (oData.length > 0 && oDetailPage) {
+          let firstDataset = oData[0];
+          this.getRun(oDetailPage, oTable, firstDataset.datasetName, firstDataset.datasetVersion, firstDataset.runId)
+        }
+      },
+      () => {
+        sap.m.MessageBox
+          .error("Failed to get the list of runs. Please wait a moment and try reloading the application")
+      })
+  };
+
+  this.getDatasetRuns = function (oControl, sDatasetName, sDatasetVersion) {
+    Functions.ajax("api/runs/" + encodeURI(sDatasetName) + "/" + encodeURI(sDatasetVersion), "GET", {},
+      oData => {
+        this._bindRunSummaries(oData, oControl);
+      },
+      () => {
+        sap.m.MessageBox
+          .error("Failed to get the list of runs for '" + sDatasetName + " (v" + sDatasetVersion +
+            ")'. Please wait a moment and try reloading the application")
+      })
+  };
+
+  this.getRun = function (oControl, oTable, sDatasetName, sDatasetVersion, sRunId) {
+    Functions.ajax("api/runs/" + encodeURI(sDatasetName) + "/" + encodeURI(sDatasetVersion) + "/" + encodeURI(sRunId), "GET", {},
+      oData => {
+        this.setCurrentRun(oControl, oTable, oData);
+      },
+      () => {
+        sap.m.MessageBox
+          .error("Failed to get the list of runs. Please wait a moment and try reloading the application")
+      })
+  };
+
+  this.getLatestRun = function (oControl, oTable, sDatasetName, sDatasetVersion) {
+    Functions.ajax("api/runs/" + encodeURI(sDatasetName) + "/" + encodeURI(sDatasetVersion) + "/latest", "GET", {},
+      oData => {
+        this.setCurrentRun(oControl, oTable, oData);
+      },
+      () => {
+        sap.m.MessageBox
+          .error("Failed to get the list of runs. Please wait a moment and try reloading the application")
+      })
+  };
+
+  this.getLatestRunForLatestVersion = function (oControl, oTable, datasetName) {
+    Functions.ajax("api/runs/" + encodeURI(datasetName) + "/latest", "GET", {},
+      oData => {
+        this.setCurrentRun(oControl, oTable, oData);
+      },
+      () => {
+        sap.m.MessageBox
+          .error("Failed to get the list of runs. Please wait a moment and try reloading the application")
+      })
+  };
+
+  this.setCurrentRun = function (oControl, oTable, oRun) {
+    let checkpoints = oRun.controlMeasure.checkpoints;
+
+    this._preprocessRun(oRun, checkpoints);
+    this._setUpCheckpointsTable(checkpoints, oTable);
+
+    oControl.setModel(new sap.ui.model.json.JSONModel(oRun), "run");
+    oControl.setModel(new sap.ui.model.json.JSONModel(oRun.controlMeasure.metadata), "metadata");
+  };
+
+  this._bindRunSummaries = function(oRunSummaries, oControl) {
+    oRunSummaries.forEach(run => {
+      run.status = this._normalizeStatus(run.status)
+    });
+    oRunSummaries.sort((a, b) => b.runId - a.runId);
```

Comment:
please sort by the start time. Now it looks like this and it is still a bit hard to understand. Even more, since each version has it's own id's. So something run for the first time would end on the end.

![image](https://user-images.githubusercontent.com/17467768/54769314-405c9b80-4c01-11e9-83ad-c4a7148c83ac.png)

Reply:
Will be solved in later PR, which incorporates a more complex menu of runs.

Reply:
#273